sdk/python/feast/feature_store.py

```diff
@@ -1270,19 +1276,22 @@ def _set_table_entity_keys(
 
     @staticmethod
     def _populate_request_data_features(
+        online_features_response: GetOnlineFeaturesResponse,
         request_data_features: Dict[str, List[Any]],
-        result_rows: List[GetOnlineFeaturesResponse.FieldValues],
     ):
         # Add more feature values to the existing result rows for the request data features
         for feature_name, feature_values in request_data_features.items():
             proto_values = python_values_to_proto_values(
                 feature_values, ValueType.UNKNOWN
             )
 
+            online_features_response.metadata.feature_names.val.append(feature_name)
```

Comment:
Why do we put values from request back into response? It seems inconsistent with the fact that we don't put entities in response

Reply:
Same reason as for entities